frontend/src/components/pages/Default.tsx

```diff
@@ -60,9 +110,12 @@ const Default = (): React.ReactElement => {
           width="100%"
           spacing={5}
         >
-          {courses?.map((course: CourseResponse) => (
-            <CourseCard key={course.id} course={course} />
-          ))}
+          {courses?.map((course: CourseResponse) => {
+            if (checkFilter(selectedTab, course.id)) {
+              return <CourseCard key={course.id} course={course} />;
+            }
+            return <></>;
```

Comment:
Might be good to return `null` instead to avoid creating extra nodes.

Reply:
I believe react fragments don't result in any dom elements. So, this should be equivalent, right?

Reply:
Technically I think it creates a virtual DOM node in React, but it probably doesn't matter practically, I guess.